File: ios/StepsCounterApp/WidgetUpdater.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WidgetUpdater, NSObject)

RCT_EXTERN_METHOD(reloadAllWidgets)
RCT_EXTERN_METHOD(reloadStepsWidgets)

+ (BOOL)requiresMainQueueSetup
{
    return NO;
}

@end 

File: ios/StepsCounterApp/WidgetUpdater.swift
import Foundation
import WidgetKit

@objc(WidgetUpdater)
class WidgetUpdater: NSObject {
    
    @objc
    static func moduleName() -> String! {
        return "WidgetUpdater"
    }
    
    @objc
    func reloadAllWidgets() {
        DispatchQueue.main.async {
            WidgetCenter.shared.reloadAllTimelines()
        }
    }
    
    @objc
    func reloadStepsWidgets() {
        DispatchQueue.main.async {
            WidgetCenter.shared.reloadTimelines(ofKind: "StepsWidgetSmall")
            WidgetCenter.shared.reloadTimelines(ofKind: "StepsWidgetMedium")
        }
    }
} 